pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
When I originally wrote this, I had it the way you want, but then I switched it the way it is right now for this reason. AFAICT, it was pretty rare that the watch closed while the resource version was still usable, so it didn't seem important. Is this not true?